crypto/crypto/src/error.rs

```diff
@@ -0,0 +1,6 @@
+// Copyright (c) The Libra Core Contributors
```

Comment:
why introduce this abstraction?

Reply:
I see you found one of the odder changes I had to make. This all boils down to the fact that the type `failure::Error` is a part of the public API of the crypto crate for a number of traits. We also have some derives in the crypto crate which expect that the failure crate exists in the global scope, since I'm trying to remove the failure crate I needed to break that assumption and the easiest way is to just re-export the error type here and then point the derives at this type instead of the `failure::Error` one.